pkg/spdk/engine.go

```diff
@@ -1797,6 +1805,62 @@ func (e *Engine) BackupRestore(spdkClient *spdkclient.Client, backupUrl, engineN
 	return resp, nil
 }
 
+func (e *Engine) completeBackupRestore(spdkClient *spdkclient.Client) error {
+	if err := e.waitForRestoreComplete(); err != nil {
+		return errors.Wrapf(err, "failed to wait for restore complete")
+	}
+
+	return e.BackupRestoreFinish(spdkClient)
```

Comment:
If there is a goroutine wait here then automatically invoke `e.BackupRestoreFinish`, do we still need to expose gRPC API `EngineBackupRestoreFinish`?

Besides, I just notice that `e.BackupRestoreFinish` will attach all replicas and create a RAID (but without any further operations like checksum verification). What is the purpose? In other words, do we really need `e.BackupRestoreFinish`?

Reply:
> If there is a goroutine wait here then automatically invoke `e.BackupRestoreFinish`, do we still need to expose gRPC API `EngineBackupRestoreFinish`?
> 

After discussing with @derekbit , we agreed that we don't need to expose gRPC API `EngineBackupRestoreFinish`, as it's not currently used by either v1 or v2 engine. Originally, this API was to abstract the finish operation.

> Besides, I just notice that `e.BackupRestoreFinish` will attach all replicas and create a RAID (but without any further operations like checksum verification). What is the purpose? In other words, do we really need `e.BackupRestoreFinish`?

At the start of the backup restore operation, we deleted the RAID bdev, disconnected all replicas, and set the e.IsRestoring flag. The purpose of `e.BackupRestoreFinish` is to complete the cycle by reattaching the replicas, recreating the RAID bdev, and resetting the flag.

Reply:
[Chore: remove EngineBackupRestoreFinish RPC #25](https://github.com/longhorn/types/pull/25).

Reply:
I see. Although launching a RAID for a restoring/DR volume is redundant, we don't need to modify the periodic verification logic.